module/updatable_configs/require_approvals_for_seal_construction.go

```diff
@@ -0,0 +1,60 @@
+package updatable_configs
+
+import (
+	"fmt"
+
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/updatable_configs/validation"
+)
+
+// min number of approvals required for constructing a candidate seal
+type requiredApprovalsForSealConstructionInstance struct {
+	requiredApprovalsForSealConstruction *atomic.Uint32
+	requiredApprovalsForSealVerification uint
+	chunkAlpha                           uint
+}
+
+func NewRequiredApprovalsForSealConstructionInstance(
+	requiredApprovalsForSealConstruction uint,
+	requiredApprovalsForSealVerification uint,
+	chunkAlpha uint,
+) (module.RequiredApprovalsForSealConstructionInstanceSetter, error) {
+	err := validation.ValidateRequireApprovals(
+		requiredApprovalsForSealConstruction,
+		requiredApprovalsForSealVerification,
+		chunkAlpha,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("can not create RequiredApprovalsForSealConstructionInstance: %w", err)
+	}
+	return &requiredApprovalsForSealConstructionInstance{
+		requiredApprovalsForSealConstruction: atomic.NewUint32(uint32(requiredApprovalsForSealConstruction)),
+		requiredApprovalsForSealVerification: requiredApprovalsForSealVerification,
+		chunkAlpha:                           chunkAlpha,
+	}, nil
+}
+
+// SetValue updates the requiredApprovalsForSealConstruction and return the old value
+// This assume the caller has validated the new value
+func (r *requiredApprovalsForSealConstructionInstance) SetValue(requiredApprovalsForSealConstruction uint) (uint, error) {
```

Comment:
If the value is invalid, yes, it won't crash.

To me, either to crash is determined by the caller, not this module itself. But the module is responsible to help caller distinguish between normal error and exceptional error. 

If this module involves a database call, then we need to use sentinel error type to distinguish between data not found, and database exceptional error. 

However, this method is doing a pure computation, meaning there is no side-effect, therefore no exceptional error. 

If there is no exception error to distinguish from, I usually don't bother to create a sentinel error, just create an error with `fmt.Errorf` and return it. 

If the method only returns one type of error, then I usually don't specifically mention about the error type, but just saying it will return error. 

I'm not sure how to apply the default convention "(any error leads to component or node crash unless explicitly specified otherwise)." Do you mean we should make a sentinel error even if there is only one type of error?

Reply:
> this method is doing a pure computation, meaning there is no side-effect, therefore no exceptional error.
> If the method only returns one type of error, then I usually don't specifically mention about the error type, but just saying it will return error.

I am not trying to convince you to use sentinel errors. I am requesting that you document exactly what you explained here in your comment. **Please clearly document that any error is equivalent to a no-op.** (so other engineers don't have to reverse-engineer this insight from the code)